python/lsst/ts/externalscripts/base_parameter_march.py

```diff
@@ -0,0 +1,461 @@
+        await asyncio.sleep(1.0)
+        await self.tcs.start_tracking()
+
+    async def parameter_march(self) -> None:
+        """Perform the parameter_march operation."""
+
+        start_position = self.config.step_sequence[0]
+
+        offset_values = start_position * self.dofs
+        cam_hex, m2_hex, m1m3_bend, m2_bend = await self.format_values(offset_values)
+
+        await self.checkpoint(
+            f"Step 1/{self.config.n_steps} starting positions:\n"
+            f"Cam Hexapod: {cam_hex}\n"
+            f"M2 Hexapod: {m2_hex}\n"
+            f"M1M3 Bend: {m1m3_bend}\n"
+            f"M2 Bend: {m2_bend}."
+        )
+        self.log.info("Offset dofs to starting position.")
+
+        # Apply dof vector with offset
+        offset_dof_data = await self.tcs.rem.mtaos.cmd_offsetDOF.DataType()
+        for i, dof_offset in enumerate(offset_values):
+            offset_dof_data.value[i] = dof_offset
+        await self.tcs.rem.mtaos.cmd_offsetDOF.start(data=offset_dof_data)
+        self.total_offset += start_position
+
+        self.iterations_started = True
+
+        # Move rotator
+        await self.track_target_with_rotation(self.config.rotation_sequence[0])
+
+        await self.take_images()
+
+        for self.iterations_executed in range(1, self.config.n_steps):
+            await self.checkpoint(
+                f"Step {self.iterations_executed+1}/{self.config.n_steps}."
+            )
+            # Calculate the offset for the current step
+            offset = (
+                self.config.step_sequence[self.iterations_executed]
+                - self.config.step_sequence[self.iterations_executed - 1]
+            )
+
+            # Apply dof vector with offset
+            offset_dof_data = await self.tcs.rem.mtaos.cmd_offsetDOF.DataType()
+            for i, dof_offset in enumerate(self.dofs * offset):
+                offset_dof_data.value[i] = dof_offset
+            await self.tcs.rem.mtaos.cmd_offsetDOF.start(data=offset_dof_data)
+
+            # Store the total offset
+            self.total_offset += offset
+
+            # Move rotator
+            await self.track_target_with_rotation(
+                self.config.rotation_sequence[self.iterations_executed]
+            )
+
+            # Take images at the current dof position
+            await self.take_images()
+
+    @abc.abstractmethod
+    def get_instrument_configuration(self) -> dict:
+        """Get the instrument configuration.
+
+        Returns
+        -------
+        dict
+            Dictionary with instrument configuration.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_filter(self) -> str:
+        """Get the instrument filter configuration.
+
+        Returns
+        -------
+        str
+            Instrument filter configuration.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def take_images(self) -> None:
```

Comment:
Can't you make `take_images` be the same for both cameras using `self.camera`?

Reply:
well, for ComCam take images are triplets, while for LSSTCam take images is only taking on image. That's why I had separated. Additionally, in the case of ComCam we ping RA to start processing the data

Reply:
Is LSSTCam really taking only one image? I think the idea here would be to do intra/extra regardless no?

Reply:
Well, LSSTCam gets intra focal and extra focal images directly from the corners for free, so we don't need to move the focal plane, unless we are using the "full array mode"